website/client/src/components/userMenu/profile.vue

```diff
@@ -836,8 +841,23 @@ export default {
       const profileUserId = this.userId;
 
       if (profileUserId && profileUserId !== this.userLoggedIn._id) {
-        const response = await this.$store.dispatch('members:fetchMember', { memberId: profileUserId });
-        user = response.data.data;
+        try {
+          const response = await this.$store.dispatch('members:fetchMember', {
+            memberId: profileUserId,
+          });
+          // debugger;
+          user = response.data.data;
+        } catch (e) {
+          // user = null;
+          user.active = false;
```

Comment:
In case the user cannot be fetched, why not keep `this.user` undefined and in the component above do `v-if="!user"` instead of adding a new property to the user object?

Reply:
I agree with you. I don't think I should be adding a new property to the user. However, when I try and do it your suggested way.. I keep getting this bug where the 404 will flash on the modal for every active user before their actual profile displays. Adding the new property is the only way I could get it to work without flashing 404 for active users.

I am still looking into this.

Reply:
That seems related to this issue: https://github.com/HabitRPG/habitica/issues/11051
(I.e., it's known that profiles can be slow to load and if caching of a previous profile hasn't happened, you see a blank profile screen before the data loads). That issue is tolerable - we rarely see complaints about it, but if the default state was the 404 error, I'm sure that would cause confusion and more bug reports.

So, I'm a big fan of not flashing up the 404 message initially. :)
If it tunrs out to not be possible to find a way to avoid that without adding a new properly, perhaps we could have the new property as a temporary workaround and then adjust the code later once https://github.com/HabitRPG/habitica/issues/11051 is fixed?

Reply:
Sounds good to me. Im having a hard time getting it to work correctly without adding the property. @paglias Any thoughts?

Reply:
@jacobguinther can you post the version without the property? I'll give a look and see if I can get it to work

Reply:
eh. Found another bug Im working on. 
![logged_in_user_404_bug](https://media.giphy.com/media/S6SICfPVYec0CEdIlJ/giphy.gif)
Currently. If I refresh the page the  logged in users profile modal shows up fine. However if i click any deleted user.. Now the logged in users profile shows 404 also.

I will post version without property once I fix this.

Reply:
Ok. I fixed the above bug and got it to work without adding the extra property on user object.